woocommerce/woocommerce
# Add spotlight to draw attention to background removal feature

### Changes proposed in this Pull Request:

Adding introductory spotlight for remove background featured. I ended up refactoring the tour component in this plugin to simplify things.

Closes #40495 

### Screenshots
![image](https://github.com/woocommerce/woocommerce/assets/444632/949447d8-1fb1-4f8c-97b2-69c55c04075c)



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Checkout branch
2. Go to Products -> Add new
3. Run `localStorage.setItem( 'debug', '*tracks*' );` in console.
4. Navigate to "set product image" to open Media Library modal.
5. Click on an image, and the pictured spotlight should appear
6. You should also see a corresponding `spotlight_displayed` tracks event fire.
7. Click the close button on the spotlight, and you should see another corresponding `spotlight_dismissed` tracks event fire.
8. The spotlight should not display again on subsequent clicks (Enter `localStorage.removeItem('woo-ai-plugin-prefs')` to see it appear again)



### Changelog entry




<details>

#### Significance




#### Type




#### Message Adding spotlight feature for background removal link to draw attention to feature.

#### Comment 

</details>
